force-app/main/default/classes/ADDR_Addresses_UTIL.cls

```diff
@@ -714,7 +764,7 @@ public with sharing class ADDR_Addresses_UTIL {
                 Address__c addrMatch = null;
                 for (Address__c childAddr : childAddrs) {
                     //Only include address type in the test if the new one isn't empty.
-                    if (!isAddressChanged(addr, childAddr, addr.Address_Type__c != null)) {
+                    if (!isAddressInfoSame(addr, childAddr, addr.Address_Type__c != null)) {
```

Comment:
@MatthewBlanski this is the trick i was referring to last week. Basically we will keep the same `same` comparison method so that it will consider address with different inactive status to share the same address info. Changes to inactive fields will still be recognized as we have an updated method for isAddressChanged. 

Sadly this is not confusing at all but it is how address is working right by always creating and deduplicating.